CanHazFunny/CanHazFunny.Tests/JesterTests.cs

```diff
@@ -1,10 +1,36 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using System.Diagnostics.Metrics;
 
 namespace CanHazFunny.Tests
 {
     [TestClass]
     public class JesterTests
     {
-        
+        [TestMethod]
+        public void JokeService_TellsJoke()
+        {
+            //Arrange
+            Jester jester = new Jester(new JokeService(), new JokeServiceWriter());
+
+            //Act
+            string joke = jester.TellJoke();
+
+            //Assert
+            Assert.IsNotNull(joke);
+        }
+        [TestMethod]
+        public void JokeService_DoesntTellChuckNorrisJokes_Please() 
+        {
+            //Arrange
+            Jester jester = new Jester(new JokeService(), new JokeServiceWriter());
+
+            //Act
+            string [] jokes = new string[10];
+            for(int i = 0; i < jokes.Length; i++) jokes[i] = jester.TellJoke();
+
+            //Assert
+            for (int i = 0; i < jokes.Length; i++) Assert.AreEqual(false, jokes[i].Contains("Chuck Norris", System.StringComparison.CurrentCulture));
+        }
```

Comment:
```suggestion
        public void JokeService_DoesntTellChuckNorrisJokes_Please() 
        {
            StringWriter stringWriter = new StringWriter();
            Console.SetOut(stringWriter);

            // Act
            Jester?.TellJoke();

            // Assert
            Assert.IsFalse(stringWriter.ToString().Contains("Chuck Norris"));
        }
```
There is an easier way of doing this. I'd recommend using something similar to the changes I provided. It essentially allows you to capture the unit test output and do with it as you want.

Reply:
Awesome review @StrangeRanger

Reply:
A good suggestion was made here. Plus, your test doesn't ensure "Chuck Norris" doesn't ever get returned, it just ensures the ten times you try it doesn't. Using a mocker would have been much more appropriate here.